docs/cvl/invariants.md

```diff
@@ -597,3 +597,62 @@ is always logically sound to add a `requireInvariant` to a `preserved` block,
 even for complicated interdependent invariants (as long as the required
 invariants have been verified).
 
+Practical example: solvency envelope
+------------------------------------
+
+It is common to assert a coarse solvency envelope and then use `filtered` and `preserved` to keep the proof tractable while avoiding vacuity. For an ERC-4626-style wrapper over an interest-bearing aToken, the following is effective:
+
+```cvl
+/// Total aTokens (scaled) covering totalSupply of shares
+invariant solvency_total_asset_geq_total_supply()
+    (_AToken.scaledBalanceOf(currentContract) >= totalSupply())
+    filtered { f ->
+        f.contract == currentContract
+        && !harnessMethodsMinusHarnessClaimMethods(f)
+        && !claimFunctions(f)
+        && f.selector != sig:emergencyEtherTransfer(address,uint256).selector
+        && f.selector != sig:claimDoubleRewardOnBehalfSame(address, address, address).selector } {
+
+    preserved withdraw(uint256 assets, address receiver, address owner) with (env e) {
```

Comment:
no no no no. you don't just put 50 lines of CVL without explaining every part. and I don't think these tips are even correct.